### ISSUE TITLE
Record constructor invoked with null argument although suppress(NULL_FIELDS) is configured

### ISSUE THREAD

**Issue by C-Otto** [jqno/equalsverifier#610]
**Describe the bug**
The following invocation tries to instantiate my record `Foo` with `null` parameters, which causes failures.
`EqualsVerifier.configure().suppress(NULL_FIELDS).forClass(Foo.class).usingGetClass().verify();`

It worked before I migrated `Foo` from a class to a record, i.e. the issue seems to be record specific.

I'm using https://github.com/uber/NullAway to ensure that my code is free of `null` fields/parameters/... without cluttering the code with `@Nonnull` annotations.

**To Reproduce**
```
public record Foo(String bar) {
    public Foo {
        assert bar != null;
    }
}

In test:
EqualsVerifier.configure().suppress(NULL_FIELDS).forClass(Foo.class).usingGetClass().verify();
```

**Error message**
```
Caused by: nl.jqno.equalsverifier.internal.exceptions.ReflectionException: Record: failed to invoke constructor.
	at app//nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:34)
	at app//nl.jqno.equalsverifier.internal.reflection.RecordObjectAccessor.callRecordConstructor(RecordObjectAccessor.java:140)
	at app//nl.jqno.equalsverifier.internal.reflection.RecordObjectAccessor.makeAccessor(RecordObjectAccessor.java:119)
	at app//nl.jqno.equalsverifier.internal.reflection.RecordObjectAccessor.clear(RecordObjectAccessor.java:81)
	at app//nl.jqno.equalsverifier.internal.reflection.ClassAccessor.getDefaultValuesAccessor(ClassAccessor.java:227)
	at app//nl.jqno.equalsverifier.internal.checkers.RecordChecker.check(RecordChecker.java:35)
	at app//nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.verifyWithoutExamples(SingleTypeEqualsVerifierApi.java:409)
	at app//nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.performVerification(SingleTypeEqualsVerifierApi.java:376)
	at app//nl.jqno.equalsverifier.api.SingleTypeEqualsVerifierApi.verify(SingleTypeEqualsVerifierApi.java:312)
	... 84 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at nl.jqno.equalsverifier.internal.reflection.RecordObjectAccessor.lambda$callRecordConstructor$4(RecordObjectAccessor.java:141)
	at nl.jqno.equalsverifier.internal.util.Rethrow.rethrow(Rethrow.java:30)
	... 92 more
Caused by: java.lang.AssertionError
	at somewhere.Foo.<init>(Foo.java:7)
	... 99 more
```

**Expected behavior**
Only non-`null` arguments are passed to the constructor.

**Version**
3.9

**Additional context**
Java 17

**Comment by jqno** [comment]
You're right, it fails on a record but succeeds on an equivalent class, so that's a bug. I'll look into it and let you know when I have a fix.

**Comment by jqno** [comment]
I've just released version 3.9.1 which fixes this bug!
